inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/ExtractNodeProviderFactory.java

```diff
@@ -20,54 +20,34 @@
 import org.apache.inlong.manager.common.enums.ErrorCodeEnum;
 import org.apache.inlong.manager.common.exceptions.BusinessException;
 import org.apache.inlong.manager.pojo.sort.node.base.ExtractNodeProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.HudiProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.IcebergProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.KafkaProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.MongoDBProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.MySQLBinlogProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.OracleProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.PostgreSQLProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.PulsarProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.RedisProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.SQLServerProvider;
-import org.apache.inlong.manager.pojo.sort.node.provider.TubeMqProvider;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
 
 /**
  * Factory of the extract node provider.
  */
+@Service
+@Slf4j
 public class ExtractNodeProviderFactory {
 
     /**
      * The extract node provider collection
      */
-    private static final List<ExtractNodeProvider> EXTRACT_NODE_PROVIDER_LIST = new ArrayList<>();
-
-    static {
-        // The Providers Parsing SourceInfo to ExtractNode which sort needed
-        EXTRACT_NODE_PROVIDER_LIST.add(new HudiProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new KafkaProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new MongoDBProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new OracleProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new PulsarProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new RedisProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new TubeMqProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new SQLServerProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new PostgreSQLProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new MySQLBinlogProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new IcebergProvider());
-
-    }
+    @Autowired
+    private List<ExtractNodeProvider> EXTRACT_NODE_PROVIDER_LIST = new ArrayList<>();
```

Comment:
why use upper case for EXTRACT_NODE_PROVIDER_LIST?

Reply:
Fixed.